tests/epyccel/test_epyccel_functions.py

```diff
@@ -333,23 +333,23 @@ def f20(a):
         b = a
         return b
     f = epyccel(f20, language=language)
-    assert f(complex(1, 2.2) == f20(complex(1, 2.2)))
+    assert f(complex(1, 2.2)), f20(complex(1, 2.2))
 
 def test_decorator_f21(language):
     @types('complex64')
     def f21(a):
         b = a
         return b
     f = epyccel(f21, language=language)
-    assert f(complex(1, 2.2) == f21(complex(1, 2.2)))
+    assert np.isclose(f(complex(1, 2.2)), f21(complex(1, 2.2)), rtol = 1e7, atol = 1e8)
```

Comment:
This tolerance is worryingly high. Does this break if you use the same tolerance as in `tests/epyccel/recognised_functions/test_numpy_funcs.py` ?

Reply:
yes it will break

Reply:
Do you have an explanation for why?

Reply:
real and imag part of the complex are either double or float.  so when casting the double value to float we lose precision  because the float is stored in 32 bit [1 sign 8 exponent 23 mantissa] and double is stored in 64 [1, 11,  52]

Reply:
Why are we casting to a float?

Reply:
yes why ? it was before we implementing numpy function to collect value but now that we are using memcpy it should  work, im going to simulate the behaviour to see what happen in memory.

Reply:
found it  ... here we are using complex64 and passing a python complex as arguments so it passes from the pytype_cast_function and not the numpy one.